beacon-chain/execution/service.go

```diff
@@ -155,6 +157,10 @@ type Service struct {
 	lastReceivedMerkleIndex int64 // Keeps track of the last received index to prevent log spam.
 	runError                error
 	preGenesisState         state.BeaconState
+	capabilities            []string
+	capabilitiesLock        sync.RWMutex
```

Comment:
Instead of having these two exposed and having to have the following type of logic on each call:

```
	s.capabilitiesLock.RLock()
	if !slices.Contains(s.capabilities, GetBlobsV1) {
		s.capabilitiesLock.RUnlock()
		return nil, nil
	}
	s.capabilitiesLock.RUnlock()
```
It's better to have a simple cache (you can copy any of the latest caches that are a struct with a lock inside. And you call s.capabilitiesCache.Get()  / Set() that are self locking